### PR TITLE
NAS-127428 / 24.10 / Installer HTTP API

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

**Comment by themylogin** [comment]
time 30:00

File: truenas_installer/network_interfaces.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+class NetworkInterface:
+    name: str
+
+
+async def list_network_interfaces():
+    max_retries = 3
+    for attempt in range(1, max_retries + 1):
+        try:
+            with IPRoute() as ipr:
+                interfaces = [NetworkInterface(dev.get_attr("IFLA_IFNAME")) for dev in ipr.get_links()]
```
[the remark this answers is not in the document]
Comment (themylogin) [reply truenas_installer/network_interfaces.py:18]: Yes, it does not perform any IO

**Review by sonicaj — APPROVED** [review]
Should we have a github action perhaps which can generate the documentation on merges to master ?

**Comment by themylogin** [comment]
@sonicaj can github actions commit?

**Comment by sonicaj** [comment]
@themylogin yes, please see https://github.com/truenas/charts/blob/master/.github/workflows/update_catalog.yaml

**Pull request merged**